app/src/main/java/com/example/eric/memberquiz/GameActivity.java

```diff
@@ -0,0 +1,306 @@
+
+        int new_score = pref.getInt("score", 0);
+        score.setText(Integer.toString(new_score));
+
+        Button[]answer_buttons = new Button[4];
+        answer_buttons[0] = button1;
+        answer_buttons[1] = button2;
+        answer_buttons[2] = button3;
+        answer_buttons[3] = button4;
+
+        //Clear All Button Text
+        for(int i = 0; i < answer_buttons.length; i++){
+            answer_buttons[i].setText("");
+        }
+
+        int rand_but_num = rand.nextInt(4);
+
+
+        //Gets other 3 names randomly
+        int rand_name1 = rand.nextInt(44);
+        while (rand_name1 == rand_img_num){
+            rand_name1 = rand.nextInt(44);
+        }
+        String wrong_name1 = image_names.get(rand_name1);
+
+        int rand_name2 = rand.nextInt(44);
+        while (rand_name2 == rand_img_num || rand_name2 == rand_name1){
+            rand_name2 = rand.nextInt(44);
+        }
+        String wrong_name2 = image_names.get(rand_name2);
+
+        int rand_name3 = rand.nextInt(44);
+        while (rand_name3 == rand_img_num || rand_name3 == rand_name1 || rand_name3 == rand_name2){
+            rand_name3 = rand.nextInt(44);
+        }
+        String wrong_name3 = image_names.get(rand_name3);
+
+
+        // Gets properly formatted name of the Image
+        for(int i = 0; i < members.length; i++){
+            if(members[i].replaceAll("\\s+","").equalsIgnoreCase(rand_img)){
+                rand_img = members[i];
+            }
+            if(members[i].replaceAll("\\s+","").equalsIgnoreCase(wrong_name1)){
+                wrong_name1 = members[i];
+            }
+            if(members[i].replaceAll("\\s+","").equalsIgnoreCase(wrong_name2)){
+                wrong_name2 = members[i];
+            }
+            if(members[i].replaceAll("\\s+","").equalsIgnoreCase(wrong_name3)){
+                wrong_name3 = members[i];
+            }
+        }
+
+        String[]button_options = new String[3];
+        button_options[0] = wrong_name1;
+        button_options[1] = wrong_name2;
+        button_options[2] = wrong_name3;
+
+        // Set Button Texts
+        answer_buttons[rand_but_num].setText(rand_img);
+        for(int i = 0, j = 0; i < answer_buttons.length && j < button_options.length; i++){
+            if (answer_buttons[i].getText() == ""){
+                answer_buttons[i].setText(button_options[j]);
+                j++;
+            }
+        }
+
+        //Count Down Timer
+        timer = new CountDownTimer(5500, 1000) {
+
+            public void onTick(long millisUntilFinished) {
+                countdownText.setText("seconds remaining: " + millisUntilFinished / 1000);
+            }
+
+            public void onFinish() {
+                countdownText.setText("Times Up!");
+                if (countdownText.getText().toString().equals("Times Up!")) {
+                    finish();
+                    startActivity(getIntent());
```

Comment:
This is a big issue because it opens a new screen every time instead of updating the screen. Imagine if Facebook reloaded the whole website everytime someone posted something. This brings up huge issues, such as adding back button functionality, ruining the UX, and generally being slow and dangerous to the OS.